examples/lfric/scripts/rm_hex.py

```diff
@@ -0,0 +1,175 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2022, Science and Technology Facilities Council
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: R. W. Ford, STFC Daresbury Laboratory
+
+'''File containing a PSyclone transformation script for the Dynamo0.3
+API which removes specified halo exchanges associated with a field
+being accessed within a kernel. Use of this script is entirely at the
+risk of the user and will produce invalid code if used
+incorrectly. The reason the script is useful is because PSyclone
+assumption about continuous fields requiring a halo exchange is not
+always the case (due to the way kernels are written internally).
+
+'''
+from psyclone.psyGen import Kern, HaloExchange
+from psyclone.psyir.transformations import TransformationError
+
+# Add all selected invoke/kernel, position/arg names here. The kernel
+# information has a name and a position, with the position being the
+# nth time this kernel is found in a schedule, as a kernel may be used
+# multiple times in a schedule.
+invoke_info = {
```

Comment:
Presumably the idea is that there will be a copy of this script for each algorithm that needs modifying. It would therefore be better if the functionality was separate to the specification of the halo exchanges to delete. However, this is only an example so it can stay as is.